Mapping/ClassMetadata.php

```diff
@@ -49,25 +44,19 @@ class ClassMetadata extends MergeableClassMetadata
     protected $extendedClass;
 
     /**
-     * Set the URL schema to use for the subject class.
-     *
-     * e.g. {foobar}/articles/{date}
-     *
-     * @param string $schema
+     * @var AutoRouteDefinition[]
      */
-    public function setUriSchema($schema)
-    {
-        $this->uriSchema = $schema;
-    }
+    protected $definitions = array();
 
     /**
-     * Return the URL schema.
+     * Add a new auto route definition for this class.
      *
-     * @return string
+     * @param string              $name
+     * @param AutoRouteDefinition $definition
      */
-    public function getUriSchema()
+    public function setAutoRouteDefinition($name, AutoRouteDefinition $definition)
```

Comment:
It is not necessarily an addition however, if the name already exists it is a replacement, so I think `set` is better. I usually think of `$this->foo[] = $something` as `add` and `$this->foo[$name]` as `set`